frontends/common/constantFolding.h

```diff
@@ -23,36 +23,81 @@ limitations under the License.
 #include "frontends/p4/typeChecking/typeChecker.h"
 
 namespace P4 {
-// Can also be run before type checking, so it only uses types if
-// they are available.
+
+/** Implements a pass that statically evaluates many constant
+ * expressions.
+ *
+ * The pass can be invoked either with or without the reference and
+ * type maps. When type information is not available, constant folding
+ * is not done for some IR nodes.
+ *
+ * @pre: None
+ *
+ * @post: Ensures that
+ *    - some constant expressions are eliminated (TODO: specify?)
+ * 
+ *    - no operations that involve only int-typed values; at least one
```

Comment:
It is a paraphrase from something you wrote...

> The spec also indicates that operations on int-typed objects (InfInt) are done using infinite precision, producing int results. The constant folding does these operations. After constant folding there are no operations which involve only int-typed values, at least one operand has a known width, which is used to cast the other operand. 

How do we make it accurate?